Src/IronPython/Runtime/MemoryView.cs

```diff
@@ -265,12 +265,22 @@ public sealed class MemoryView : ICodeFormattable, IWeakReferenceable {
         }
 
         private byte[] unpackBytes(string format, object o) {
-            if (TypecodeOps.TryGetBytes(format, o, out byte[] bytes)) {
-                return bytes;
-            } else if (o is Bytes b) {
-                return b._bytes; // CData returns a bytes object for its type
-            } else {
-                throw PythonOps.NotImplementedError("No conversion for type {0} to byte array", PythonOps.GetPythonTypeName(o));
+            try {
```

Comment:
Looking at usage in `setByteRange`/`getByteRange` this is called with `unpackBytes(_buffer.Format, _buffer.GetItem(index)`. Do we even need all this error checking in this case?

Reply:
You're right; the error checking should probably be done in `setAtFlatIndex` because that's the place where all of the external element setting eventually calls into